client/src/hooks/useAccount.ts

```diff
@@ -24,7 +29,32 @@ const useAccount = () => {
   };
 
   const handleAccount = (event: React.ChangeEvent<HTMLInputElement>) => {
-    setAccountNumber(event.target.value);
+    if (isPasting) return;
+
+    const newValue = event.target.value;
+    const {isValid, errorMessage} = validateAccountNumber(newValue);
+    setAccountNumberErrorMessage(errorMessage);
+
+    const isValidMinLength = newValue.length >= RULE.minAccountNumberLength;
+
+    if (isValid) {
+      setAccountNumber(event.target.value);
+    } else if (!isValid && !isValidMinLength) {
+      setAccountNumber(event.target.value.replace(/[^0-9\s\-]/g, '').trim());
```

Comment:
어차피 허용되는 문자인데 slice 를 사용하면 안되는 이유가 있었을까요?
```suggestion
      setAccountNumber(event.target.value.slice(0, RULE.maxAccountNumberLength).trim());
```

Reply:
8자보다 작을 때 숫자 공백 하이픈 외의 문자가 허용되지 않도록 하기 위해서입니다!